### PR TITLE
[ci-visibility] Add Agentless documentation for CI Visibility Tests

### PR DESCRIPTION
**Pull request by drodriguezhdez**
### What does this PR do?

This PR adds the agentless documentation for CI Visibility Tests

### Motivation

The libraries to report test data to Datadog using CI Visibility supports now the agentless mode.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_tests/agentless.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  - link: "/continuous_integration/setup_tests/"
+    tag: "Next step"
+    text: "Instrumenting tests in your language"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+If you are using a SaaS CI provider with no access to the underlying worker nodes, you can send the collected data by the Test instrumentation libraries using the Agentless mode.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agentless.md:17]:
```suggestion
If you are using a SaaS CI provider without access to the underlying worker nodes, you can send the collected data by the Test instrumentation using the Agentless mode.
```
I'd say "test instrumentation libraries" does not quite fit

File: content/en/continuous_integration/setup_tests/agent.md [hunk 2]
```diff
@@ -14,12 +14,9 @@ further_reading:
...
 {{< /site-region >}}
 
-To report test results to Datadog, the [Datadog Agent][1] is required.
+To report test result to Datadog, you can do it through the [Datadog Agent][1] (recommended for on-premises installations).
 
-There are two ways to set up the Agent in a CI environment:
-
-* Install the Agent as a long running process on each worker node (recommended for on-premises installations).
-* Run an ephemeral instance of the Agent as a service container on each build (recommended for SaaS CI providers).
+For SaaS CI providers, Datadog recommends using the [Agentless mode][7] of the Test Instrumentation Libraries. This mode allows sending data without the need of running an ephemeral instance of the Agent as a service container on each build.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agent.md:19]:
```suggestion
For SaaS CI providers, Datadog recommends using the [Agentless mode][7] of the Datadog Libraries. This mode allows sending data without running an ephemeral instance of the Agent as a service container on each build.
```
I'm not convinced by "test instrumentation libraries". Also some language simplification would be useful IMO

**Review by juan-fernandez — COMMENTED** [review]
I like the direction!

File: content/en/continuous_integration/setup_tests/agentless.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+**Default**: `datadoghq.com`<br/>
+**Selected site**: {{< region-param key="dd_site" code="true" >}}
+
+### Compatibility
+
+| Library         | Support type    | Version |
+|-----------------|-----------------|---------|
+| [.NET][4]       | Beta            | x.x.x   |
+| [Java][5]       | Not supported   | x.x.x   |
+| [JavaScript][6] | Beta            | x.x.x   |
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agentless.md:44]:
```suggestion
| [JavaScript][6] | Beta            | 2.5.0   |
```

File: content/en/continuous_integration/setup_tests/agent.md [hunk 4]
```diff
@@ -14,12 +14,9 @@ further_reading:
 <div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
 {{< /site-region >}}
 
-To report test results to Datadog, the [Datadog Agent][1] is required.
+To report test result to Datadog, you can do it through the [Datadog Agent][1] (recommended for on-premises installations).
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agent.md:17]: I think on-premise is more widely used than on-premises
```suggestion
To report test results to Datadog, you can do it through the [Datadog Agent][1] (recommended for on-premise installations).
```

File: content/en/continuous_integration/setup_tests/agentless.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  - link: "/continuous_integration/setup_tests/"
+    tag: "Next step"
+    text: "Instrumenting tests in your language"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+If you are using a SaaS CI provider without access to the underlying worker nodes, you can send the collected data by the Test instrumentation using the Agentless mode.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agentless.md:17]:
```suggestion
If you are using a SaaS CI provider without access to the underlying worker nodes, you can send the collected data by the test instrumentation using the Agentless mode.
```

File: content/en/continuous_integration/setup_tests/agentless.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+For On-Premises CI providers, Datadog recommends sending data through the [Datadog Agent][1].
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agentless.md:19]:
```suggestion
For On-Premise CI providers, Datadog recommends sending data through the [Datadog Agent][1].
```

File: content/en/continuous_integration/setup_tests/agentless.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+| Library         | Support type    | Version |
+|-----------------|-----------------|---------|
+| [.NET][4]       | Beta            | 2.5.1   |
+| [Java][5]       | Beta            | 0.101.0 |
+| [JavaScript][6] | Beta            | 2.5.0   |
+| [Python][7]     | Not supported   | N/A     |
+| [Ruby][8]       | Not supported   | N/A     |
+
+For other languages like [Swift][9] or other ways of reporting test data such as [JUnit Report Upload][10], there is nothing additional to do as the only supported mode for them is agentless.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/agentless.md:48]:
```suggestion
For other languages like [Swift][9] or other ways of reporting test data such as [JUnit Report Upload][10], there is nothing additional to do as the only supported mode for them is Agentless.
```

File: content/en/continuous_integration/setup_tests/dotnet.md [hunk 8]
```diff
@@ -27,11 +27,10 @@ Supported test frameworks:
...
 ## Prerequisites
 
-[Install the Datadog Agent to collect tests data][1].
+To report test results to Datadog, you have the following options:
 
-<div class="alert alert-warning">
-Agentless mode is in beta. To test this feature, follow the <a href="/continuous_integration/setup_tests/dotnet#agentless-beta">instructions</a> on this page.
-</div>
+* If you run your tests in SaaS CI providers, Datadog recommends using the [Agentless mode][11] of the Datadog Libraries. This mode allows sending data without running an ephemeral instance of the Agent as a service container on each build.
+* If you run your tests in On-Premises CI providers, Datadog recommends sending the data through the [Datadog Agent][1].
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/dotnet.md:33]:
```suggestion
* If you run your tests in On-Premise CI providers, Datadog recommends sending the data through the [Datadog Agent][1].
```

**Review by juan-fernandez — APPROVED** [review]
LGTM

**Review by tonyredondo — APPROVED** [review]

**Review by nachoBonafonte — APPROVED** [review]

**Review by fermayo — COMMENTED** [review]
I have opened a PR (to be merged on this branch) with a totally different proposal: https://github.com/DataDog/documentation/pull/14117

WDYT?

**Comment by drodriguezhdez** [comment]
@fermayo It works for me! Thanks for the proposal 👌🏼. I'll take a final look at the PR and I'll send it to the docs team. Thanks all!

**Review by juan-fernandez — APPROVED** [review]
new proposal lgtm too 👍

File: content/en/continuous_integration/setup_tests/_index.md [hunk 9]
```diff
@@ -3,13 +3,7 @@ title: Set up Tracing on CI Tests
 kind: documentation
 ---
 
-To gather test suite results, performance, and reliability data, first set up the Agent, and then instrument your tests:
-
-{{< whatsnext desc="1. Setting up and configuring the Agent:" >}}
-    {{< nextlink href="continuous_integration/setup_tests/agent" >}}Installing the Datadog Agent{{< /nextlink >}}
-    {{< nextlink href="continuous_integration/setup_tests/containers" >}}Forwarding Environment Variables for Tests in Containers{{< /nextlink >}}
```
Comment (jtappa) [inline content/en/continuous_integration/setup_tests/_index.md:10]: It looks like the link for the Containers was removed here, but the page still exists, was this on purpose?

**Comment by drodriguezhdez** [comment]
@jtappa all the comments have been addressed. Could you take another look? Thanks!

**Review by jtappa — APPROVED** [review]

**Pull request merged**